File: DSAlgorithms/Leetcode/ts/src/2_pointers/1968-not_average.ts
const rearrangeArray = (nums: number[]): number[] =>{
    nums.sort((a, b) => a - b);
    const result: number[] = [];
    let left = 0;
    let right = nums.length - 1;
    while(result.length != nums.length)
    {
        result.push(nums[left]);
        left++;
        if(left <= right)
        {
            result.push(nums[right]);
            right--;
        }
    }
    return result;
};

File: DSAlgorithms/Leetcode/c++/2_pointers/1968-not_average.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
       sort(nums.begin(), nums.end());
       std::vector<int> result;
       int left = 0;
       int right = nums.size() - 1;
       while(result.size() != nums.size())
       {
           result.push_back(nums[left]);
           left++;
           if(left <= right)
           {
               result.push_back(nums[right]);
               right--;
           }
       }
       return result;
    }
};